File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Effect/Combat/PowerDeBuffEffect.ts
import { FairyGUI } from "csharp";
import { View_PowerDebuffEffect } from "../../Gen/View/ModTheSpire_Effect";
import { AbstractEffect } from "../AbstractEffect";

export class PowerDebuffEffect extends AbstractEffect{
    public constructor(public X: number,public Y:number,public Msg:string) {
        super();
        let effect = <FairyGUI.GComponent>View_PowerDebuffEffect.CreateInstance();
        effect.text = Msg;
        effect.SetPosition(X,Y,0);
        let transition = effect.GetTransition("t0");
        transition.Play(1,0,()=>{
            this.IsDone = true;
        });
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Effect/Combat/PowerDebuffEffect.cs
using ModTheSpire.ModTheSpire_Effect;

namespace ModTheSpire.Effect.Combat
{
    public class PowerDebuffEffect : AbstractEffect
    {
        private View_PowerDebuffEffect mEffect;
        public PowerDebuffEffect(float x, float y, string msg)
        {
            mEffect = View_PowerDebuffEffect.CreateInstance();
            mEffect.text = msg;
            mEffect.SetPosition(x, y, 0);
            mEffect.GetTransition("t0").Play(1, 0, Finish);
        }

        public override void Dispose()
        {
            base.Dispose();
            mEffect.Dispose();
        }
    }
}